File: dotNET/PFProSDK/src/PayPal/Payments/DataObjects/ExpressCheckoutRequest.cs
#region "Imports"

using System;
using PayPal.Payments.DataObjects ;
using PayPal.Payments.Common.Utility; 
#endregion

namespace PayPal.Payments.DataObjects
{
	/// <summary>
	/// This  class serves as base class of all ExpressCheckout request classes.
	/// </summary>
	/// <remarks>
	/// <para>Each request object is associated with a particular type of expressChecout operation.</para>
	/// <para>Following are the request objects associated with 
	/// different operations of ExpressChecout:</para>
	/// <list type="table">
	/// <listheader>
	/// <term>ExpressCheckout operation.</term>
	/// <description>Request data object</description>
	/// </listheader>
	/// <item>
	/// <term>SET operation for ExpressCheckout.</term>
	/// <description><see cref="ECSetRequest">ECSetRequest</see></description>
	/// </item>
	/// <item>
	/// <term>GET operation for ExpressCheckout.</term>
	/// <description><see cref="ECGetRequest">ECGetRequest</see></description>
	/// </item>
	/// <item>
	/// <term>DO operation for ExpressCheckout.</term>
	/// <description><see cref="ECDoRequest">ECDoRequest</see></description>
	/// </item>
	/// </list>
	/// </remarks>
	public class ExpressCheckoutRequest : BaseRequestDataObject
	{
		#region "Member variables"
		private String mToken;
		private String mCountryCode;
		private String mPostalCode;
		private String mAction;
		private String mDoReauthorization;
		#endregion

		#region "Properties"
		
		/// <summary>
		/// Gets and sets the value of the token.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>TOKEN</code>
		/// </remarks>
		public String Token
		{
			get{return mToken; }
			set{mToken = value;}
		}

		/// <summary>
		/// Gets and sets the country Code.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>COUNTRYCODE</code>
		/// </remarks>
		public String CountryCode
		{
			get{return mCountryCode; }
			set{mCountryCode = value;}
		}

		/// <summary>
		/// Gets and sets the postal code.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>POSTALCODE</code>
		/// </remarks>
		public String PostalCode
		{
			get{return mPostalCode; }
			set{mPostalCode = value;}
		}

		/// <summary>
		/// Gets and sets the do reauthorization flag.
		/// </summary>
		/// <remarks>
		/// <para>Maps to Payflow Parameter:</para>
		/// <code>DOREAUTHORIZATION</code>
		/// </remarks>
		public String DoReauthorization
		{
			get{return mDoReauthorization; }
			set{mDoReauthorization = value;}
		}

		#endregion

		#region "Constructor"
		/// <summary>
		/// Constructor
		/// </summary>
		internal ExpressCheckoutRequest(String Action) 
		{
			mAction = Action;
		}

		/// <summary>
		/// Constructor
		/// </summary>
		internal ExpressCheckoutRequest(String Action, String Token) 
		{
			mAction = Action;
			mToken = Token;
		}

		#endregion

		#region "Core functions"

		/// <summary>
		/// Generates the transaction request.
		/// </summary>
		internal new virtual void GenerateRequest()
		{
			// This function is not called. All the
			// address information is validated and generated
			// in its respective derived classes.
			base.GenerateRequest ();

			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_TOKEN, mToken));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_COUNTRYCODE, mCountryCode));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_POSTALCODE, mPostalCode));
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_ACTION, mAction));	
			RequestBuffer.Append(PayflowUtility.AppendToRequest(PayflowConstants.PARAM_DOREAUTHORIZATION, mDoReauthorization));
		}
		#endregion
	}
}

File: java/src/sdk/dataobjects/paypal/payflow/ExpressCheckoutRequest.java
package paypal.payflow;

/**
 * This  class serves as base class of all ExpressCheckout request classes.
 * <p>Each request object is associated with a particular type of expressChecout operation.</p>
 * <p>Following are the request objects associated with
 * different operations of ExpressCheckout:</p>
 *  * <p>
 * ExpressCheckout operation. Request data object
 * SET operation for ExpressCheckout {@link ExpressCheckoutResponse}
 * GET operation for ExpressCheckout {@link ECGetResponse}
 * DO operation for ExpressCheckout {@link ECDoResponse}
 */

public class ExpressCheckoutRequest extends BaseRequestDataObject {


    private String token;
    private String countryCode;
    private String postalCode;
    private String action;
    private String doReauthorization;

    /**
     * Gets the token parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: TOKEN
     */
    public String getToken() {
        return token;
    }

    /**
     * Sets the token parameter.
     *
     * @param token - String
     *  <p>Maps to Payflow Parameter: TOKEN
     */
    public void setToken(String token) {
        this.token = token;
    }

    /**
     * Gets the recountryCodeturnurl parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: COUNTRYCODE
     */
    public String getCountryCode() {
        return countryCode;
    }

    /**
     * Sets the countryCode parameter.
     *
     * @param countryCode - String
     *  <p>Maps to Payflow Parameter: COUNTRYCODE
     */
    public void setCountryCode(String countryCode) {
        this.countryCode = countryCode;
    }

    /**
     * Gets the postalCode parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: POSTALCODE
     */
    public String getPostalCode() {
        return postalCode;
    }

    /**
     * Sets the postalCode parameter.
     *
     * @param postalCode - String
     *  <p>Maps to Payflow Parameter: POSTALCODE
     */
    public void setPostalCode(String postalCode) {
        this.postalCode = postalCode;
    }


    /**
     * Gets the doReauthorization parameter.
     *
     * @return - String
     *  <p>Maps to Payflow Parameter: DOREAUTHORIZATION
     */
    public String getDoReauthorization() {
        return doReauthorization;
    }

    /**
     * Sets the postalCode parameter.
     *
     * @param doReauthorization - String
     *  <p>Maps to Payflow Parameter: DOREAUTHORIZATION
     */
    public void setDoReauthorization(String doReauthorization) {
        this.doReauthorization = doReauthorization;
    }


    protected ExpressCheckoutRequest(String Action) {
        action = Action;
    }


    protected ExpressCheckoutRequest(String Action, String Token) {
        action = Action;
        token = Token;
    }


    protected void generateRequest() {
        // This function is not called. All the
        //address information is validated and generated
        //in its respective derived classes.
        super.generateRequest();
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_TOKEN, token));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_COUNTRYCODE, countryCode));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_POSTALCODE, postalCode));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_ACTION, action));
        super.getRequestBuffer().append(PayflowUtility.appendToRequest(PayflowConstants.PARAM_DOREAUTHORIZATION, doReauthorization));

    }

}
